extensions/iceberg/src/main/java/io/deephaven/iceberg/util/SchemaSpec.java

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+
+import java.util.Iterator;
+
+/**
+ * A specification for extracting the schema from a table.
+ */
+interface SchemaSpec {
+
+    /**
+     * Returns the schema for the given table based on this SchemaSpec.
+     */
+    Schema getSchema(Table table);
+
+    // Static factory methods for creating SchemaSpec instances
+    static SchemaSpec initial() {
+        return new InitialSchemaSpec();
+    }
+
+    static SchemaSpec fromSchemaId(final int id) {
+        return new IdSchemaSpec(id);
+    }
+
+    static SchemaSpec fromSchema(final Schema schema) {
+        return new DirectSchemaSpec(schema);
+    }
+
+    static SchemaSpec fromSnapshotId(final int snapshotId) {
+        return new SnapshotIdSchemaSpec(snapshotId);
+    }
+
+    static SchemaSpec latest() {
+        return new LatestSchemaSpec();
+    }
```

Comment:
Looking at the spec, it looks like the preferred terminology is "current"; I would use that here.